File: benches/hashset.cpp
#include <unordered_set>
#include <iostream>
#include <cstdint>
#include <ranges>

struct MyRand {
    uint64_t seed = 0;

    uint64_t next() {
        seed = seed * 123456789 + 101112131415;
        return seed;
    }
};

using namespace std;

int main() {
    MyRand rand{};
    unordered_set<uint64_t> hashset;
    for (int _: views::iota(0, 1'000'000)) {
        hashset.insert(rand.next());
    }
    cout << hashset.size() << endl;
}

File: benches/hashset.zig
const std = @import("std");

const MyRand = struct {
    var seed: u64 = 0;

    fn next() u64 {
        seed = seed *% 123456789 +% 101112131415;
        return seed;
    }
};

pub fn main() !void {
    const HashSet = std.AutoHashMapUnmanaged(u64, void);
    var hashset = HashSet.empty;
    for (0..1_000_000) |_| {
        hashset.put(std.heap.c_allocator, MyRand.next(), void{}) catch unreachable;
    }
    var writer = std.fs.File.stdout().writer(&.{});
    writer.interface.print("{}\n", .{hashset.size}) catch unreachable;
}
